src/tikv.md

```diff
@@ -0,0 +1,212 @@
+# TiKV
+
+
+## TODOs
+- [ ] Add grpc entry path details
+- [ ] Add async-io execution path details
+- [ ] Add new request tracker related information
+- [ ] Add raft-engine execution details
+- [ ] Add kv engine details to the important stages such as `append_log`
+
+## Grpc Entry
+TODO
+
+## TxnKV Read
+
+### Point/Batch Point Get
+
+```railroad
+Diagram(
+  Sequence(
+      NonTerminal("Grpc Receive"),
+  ),
+  Choice(
+    0,
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Local Read"),
+    ),
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Leader Confirm"),
+    ),
+  ),
+  Sequence(
+    NonTerminal("Write Seek For The Target Key"),
+  ),
+  Choice(
+    0,
+    NonTerminal("Load Key Value from the Key"),
+  ),
+  Choice(
+    0,
+    Comment("Load Row Value from default cf"),
+  ),
+  Sequence(
+      NonTerminal("Grpc Response"),
+  ),
+)
+```
+
+- The snapshot get duration is observed as `tikv_storage_engine_async_request_duration_seconds_bucket{type="snapshot"}`.
+- The write seek may seek kv db iterator to the target first, there could be some next operations to skip `rollback` and `lock` records.
+- The kv db write seek duration is observed as `tikv_engine_seek_micro_seconds{type="seek_max"}`.
+
+### Coprocessor Read
+
+
+```railroad
+Diagram(
+  Sequence(
+      NonTerminal("Grpc Receive"),
+  ),
+  Choice(
+    0,
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Local Read"),
+    ),
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Leader Confirm"),
+    ),
+  ),
+  Sequence(
+    NonTerminal("Cop Task Wait Schedule"),
```

Comment:
Cop tasks will be rescheduled after 1ms each time. In a busy cluster, a cop task may be waiting for scheduling many times in the execution phase.

Reply:
We may dig deeper into the cop task execution model later, I've added a TODO above.